cylc/flow/scheduler.py

```diff
@@ -1696,139 +1696,143 @@ def update_profiler_logs(self, tinit):
     async def main_loop(self) -> None:
         """The scheduler main loop."""
         while True:  # MAIN LOOP
-            tinit = time()
+            await self._main_loop()
```

Comment:
Note: view this diff with whitespace changes hidden.

I have pulled each iteration of the main loop into it's own method for easier testing.

Reply:
Could perhaps rename this method to `run_main_loop`? Or even dispense with this method and do the `while True` in `run_scheduler()`?